felix/bpf-gpl/tc.c

```diff
@@ -559,6 +559,34 @@ static CALI_BPF_INLINE void calico_tc_process_ct_lookup(struct cali_tc_ctx *ctx)
 		goto do_policy;
 	}
 
+	/* If the dest route is a blackhole route and packet is from an external source,
+	 * drop the packet, if service loop prevention is configured to either DROP or REJECT.
+	 */
+	if (cali_rt_is_blackhole(dest_rt) && !(ctx->state->flags & CALI_ST_SKIP_FIB)) {
+		if (CALI_F_FROM_HEP && !(cali_rt_lookup(&ctx->state->ip_src))) {
```

Comment:
```suggestion
	if (CALI_F_FROM_HEP && cali_rt_is_blackhole(dest_rt) && !(ctx->state->flags & CALI_ST_SKIP_FIB)) {
		if (!(cali_rt_lookup(&ctx->state->ip_src))) {
```

Why is the `!(ctx->state->flags & CALI_ST_SKIP_FIB)` important?

Reply:
We set ```CALI_ST_SKIP_FIB``` for node local DNS, which will hit the same blackhole route.

Reply:
Iirc that is only set when from WEP.